bitnami/kafka/templates/statefulset.yaml

```
@@ -181,7 +181,7 @@ spec:
               {{- if .Values.zookeeper.enabled }}
               value: {{ printf "%s%s" (include "kafka.zookeeper.fullname" .) (tpl .Values.zookeeperChrootPath .) | quote }}
               {{- else }}
-              value: {{ include "common.tplvalues.render" (dict "value" (join "," .Values.externalZookeeper.servers) "context" $) }}
+              value: {{ include "common.tplvalues.render" (dict "value" (printf "%s%s" (join "," .Values.externalZookeeper.servers) (tpl .Values.zookeeperChrootPath .)) "context" $)    }}
```

With the current approach, `zookeeperChrootPath` will only be applied to the last Zookeeper server:
```
            - name: KAFKA_CFG_ZOOKEEPER_CONNECT
              value: test1,test2,test3/path
```

When the objective should be:
```
            - name: KAFKA_CFG_ZOOKEEPER_CONNECT
              value: test1/path,test2/path,test3/path
```

To get this result, I suggest using
```suggestion
              {{- $zookeeperExternal := list }}
              {{- range .Values.externalZookeeper.servers }}
              {{- $zookeeperExternal = append $zookeeperExternal (printf "%s%s" (include "common.tplvalues.render" ( dict "value" . "context" $)) (tpl $.Values.zookeeperChrootPath $)) }}
              {{- end }}
              value: {{ join "," $zookeeperExternal | quote }}
```

---

@migruiz4 I tried your approach first, that results in:

`KAFKA_CFG_ZOOKEEPER_CONNECT: zookeeper-0.zookeeper-headless/kafka,zookeeper-1.zookeeper-headless/kafka,zookeeper-2.zookeeper-headless/kafka`

But that's not how the configuration works. It seems that kafka first splits at the first `/`, and then interpretes everything before the '/' as the server quorum, and everything after the first `/` as the path.

I had that with the current version when I tried:

```yaml
externalZookeeper:
  servers:
  - zookeeper-0.zookeeper-headless/kafka
  - zookeeper-1.zookeeper-headless/kafka
  - zookeeper-2.zookeeper-headless/kafka
```

This gives weird results:

```
Server(s): zookeeper-0.zookeeper-headless
Path:
  /kafka,zookeeper-1.zookeeper-headless
    /kafka,zookeeper-2.zookeeper-headless
      /kafka
        /admin
```

In my approach, the result is correct:

```
Server(s): zookeeper-0.zookeeper-headless,zookeeper-1.zookeeper-headless,zookeeper-2.zookeeper-headless
Path:  /kafka
```

---

My bad!

I thought the syntax was: `host:port/path,host:port/path`.

Indeed, the correct syntax is `host:port,host:port,host:port/path`.

For reference: https://docs.confluent.io/platform/current/installation/configuration/broker-configs.html#brokerconfigs_zookeeper.connect